examples/platforms/simulation/radio.c

```diff
@@ -113,6 +114,17 @@ static int8_t         sCcaEdThresh = -74;
 
 static bool sSrcMatchEnabled = false;
 
+#if OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+static uint8_t sAckIeData[ACK_IE_MAX_SIZE];
+static uint8_t sAckIeDataLength = 0;
+#endif
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+static uint8_t  sCslIeHeader[IE_HEADER_SIZE] = {0x04, 0x0d};
+static uint32_t sCslSampleTime;
```

Comment:
I don't think radio needs to know sample time and CSL period. The IE is filled at `SubMac`. All radio needs to provide is callback to notify upper layer that transmission for normal frame and enh-ack has started. And make sure encryption is done after the callback. Thoughts?

Reply:
Related with https://github.com/openthread/openthread/pull/4557#discussion_r450564304

I remember I was suggested to put another sample time and CSL period in radio layer. Let's have a discussion and finally decide it.